src/inmanta/server/services/resourceservice.py

```diff
@@ -1138,3 +1139,23 @@ async def get_resources_in_version(
         )
 
         return ReturnValueWithMeta(response=dtos, links=links if links else {}, metadata=vars(metadata))
+
+    @handle(methods_v2.versioned_resource_details, env="tid")
+    async def versioned_resource_details(
+        self, env: data.Environment, version: int, rid: ResourceIdStr
+    ) -> VersionedResourceDetails:
+        resource = await data.Resource.get_one(environment=env.id, model=version, resource_id=rid)
+        if not resource:
+            raise NotFound("The resource with the given id does not exist")
+        parsed_id = Id.parse_id(resource.resource_id)
+        dto = VersionedResourceDetails(
+            resource_id=resource.resource_id,
+            resource_version_id=resource.resource_version_id,
+            resource_type=resource.resource_type,
+            agent=resource.agent,
+            id_attribute=parsed_id.attribute,
+            id_attribute_value=resource.resource_id_value,
+            version=resource.model,
+            attributes=resource.attributes,
+        )
```

Comment:
I think we should move this logic into `data.Resource`. That way it is more reusable across different API endpoints.